### DISCUSSION TITLE
Dto with form request in the same class in PATCH with optional fields

### DISCUSSION THREAD

**Discussion by OthmanAlseagh** [spatie/laravel-data#634] [Q&A]
i have this dto
class UpdateArticleData extends Data
{
    use HasFailedValidationDtoRequest;

    public function __construct(
        public Optional|string $title,
        public Optional|string $description,
        public Optional|bool $is_active,
        public Optional|UploadedFile $cover_image,
    ) {
    }

    public static function rules(): array
    {
        return [
            'title' => ['string',  'max:255'],
            'description' => ['string',  new CheckNumberOfWords(2100)],
            'is_active' => ['boolean'],
            'cover_image' => [File::types(['png', 'jpg', 'jpeg'])->max(2 * 1024)],
        ];
    }
}

it acts as form request and in the same time dto
in my case, i have patch method and the title is optional in update, but in the same time it is not nullble in database 

so the optional should remove it from the dto but the validation catch it 
when i use nallable in validation the database throw exception 

i don't want to handle it in code, is there a way i can handle it in the same dto ???

**Comment by rubenvanassche** [comment]
That's because you overwrite the rule in the rules method, Data won't make up rules automatically when you do that. Add #[max(255)] as an attribute to title and it should work.
